st2actions/st2actions/container/base.py

```diff
@@ -89,7 +90,9 @@ def dispatch(self, liveaction_db):
 
         return liveaction_db.result
 
-    def _do_run(self, runner, runnertype_db, action_db, liveaction_db):
+    @CounterWithTimer(key="st2.action.executions")
+    def _do_run(self, runner, runnertype_db, action_db, liveaction_db,
+                metrics_counter_with_timer=None):
```

Comment:
Does the decorator inject `metrics_counter_with_timer` argument into the decorated function?

It looks like it's not used and if that's the case, it might perhaps make sense to add an argument to the decorator with which we can control where to and where not to inject argument to the function (it looks like there is no need to inject it here).